back/app/services/participants_service.rb

```diff
@@ -104,21 +104,38 @@ def project_participants(project)
   def project_participants_count(project)
     Rails.cache.fetch("#{project.cache_key}/participant_count", expires_in: 1.day) do
       participant_ids = projects_participants([project]).pluck(:id)
-      participant_author_hashes = participant_ids.map { |id| Idea.create_author_hash(id, project.id, true) }
-      anonymous_idea_hashes = Idea.where(project: project, anonymous: true)
-        .where.not(author_hash: participant_author_hashes)
-        .distinct.pluck(:author_hash)
-      anonymous_comment_hashes = Comment.joins(:idea)
-        .where(anonymous: true)
-        .where(idea: { project: project })
-        .where.not(author_hash: participant_author_hashes + anonymous_idea_hashes)
-        .distinct.pluck(:author_hash)
-      everyone_survey_count = Idea.where(project: project, author: nil, anonymous: false).count
-
-      participant_ids.size + anonymous_idea_hashes.size + anonymous_comment_hashes.size + everyone_survey_count
+      anonymous_author_hashes = participant_ids.map { |id| Idea.create_author_hash(id, project.id, true) }
+      participant_ids.size + projects_anonymous_count([project], anonymous_author_hashes) + projects_everyone_count([project])
     end
   end
 
+  # Returns the total count of all folder participants including anonymous posts - cached
+  def folder_participants_count(folder)
+    Rails.cache.fetch("#{folder.cache_key}/participant_count", expires_in: 1.day) do
+      participant_ids = projects_participants(folder.projects).pluck(:id)
+      anonymous_author_hashes = participant_ids.flat_map do |id|
+        folder.projects.ids { |project_id| Idea.create_author_hash(id, project_id, true) }
+      end
+      participant_ids.size + projects_anonymous_count(folder.projects, anonymous_author_hashes) + projects_everyone_count(folder.projects)
+    end
+  end
+
+  def projects_anonymous_count(projects, anonymous_author_hashes)
```

Comment:
Don't think `anonymous_author_hashes` is the correct variable name here as what we are using it for is excluding authors who have not posted anonymously. Maybe a better name would be `known_author_hashes`

Reply:
Sure, I just kept the original name, but find for me to rename it to `known_author_hashes `.